synapse/util/metrics.py

```diff
@@ -62,25 +59,31 @@
 
 
 def measure_func(name=None):
-    def wrapper(func):
-        block_name = func.__name__ if name is None else name
+    """
+    Used to decorator an async function with a `Measure` context manager.
+
+    Usage:
 
-        if inspect.iscoroutinefunction(func):
+    @measure_func
```

Comment:
```suggestion
    @measure_func()
```

otherwise this does something quite different...

Reply:
Yes, good call! 👍 The functions that create decators are always a bit confusing. 😄